### PR TITLE
♻️ Refactor components and logic

### PR DESCRIPTION
**Pull request by bakseter**
- Add more eslint rules.
- Add `Props` as a type to all components.
- Enforce type checking in `getStaticProps`, `getServerSideProps`,
  etc... by checking if props passed to page component actually is of type `Props`.
- Change `||` to `??` (safer, from new eslint rule).
- Remove/add/modify various checks and conditions, from new eslint rules.
- Extract sub-components to their own file.
- Delete `useTimeout` hook (again????).
- Make 'testing-utils.js' TypeScript.
- Bump target to 'es2018'.

Kan sikkert endre på ting om det blir for sjukt.

### CODE REVIEW

**Comment by boaanes** [comment]
Shit, her må du gi meg litt tid. @alvarhonsi bør ta en titt også

File: src/components/__tests__/footer.test.tsx [hunk 1]
```diff
@@ -11,8 +11,8 @@ describe('Footer', () => {
 
     test('renders correctly', () => {
         render(<Footer />);
-        expect(screen.getByText(/Thormøhlensgate 55/i)).toBeInTheDocument();
-        expect(screen.getByText(/5006 Bergen/i)).toBeInTheDocument();
-        expect(screen.getByText(/Org nr: 998 995 035/i)).toBeInTheDocument();
+        expect(screen.getByText(/thormøhlensgate 55/i)).toBeInTheDocument();
```
Comment (bakseter) [inline src/components/__tests__/footer.test.tsx:14]: siden man bruker '/i' så ignorere man stor bokstav, fra eslint-regel

File: src/components/animated-icons.tsx [hunk 2]
```diff
@@ -13,7 +13,7 @@ const AnimatedIcons = ({ n, children }: Props): JSX.Element => {
     // Only for Halloween, no animated icons for Christmas
     if (month !== 10) return <>{children}</>;
 
-    const keys = Array.from(Array(n).keys());
+    const keys = [...new Array(n).keys()];
```
Comment (bakseter) [inline src/components/animated-icons.tsx:16]: gjør nøyaktig det samme

File: src/components/__tests__/happening-meta-info.test.tsx [hunk 3]
```diff
@@ -166,15 +166,21 @@ describe('HappeningMetaInfo', () => {
         const props = happeningMetaInfoProps[0];
         const { getByText } = render(<HappeningMetaInfo {...props} />);
 
-        expect(getByText(new RegExp(`${props.spotRangeCounts?.[0].spots} plasser`))).toBeInTheDocument();
+        expect(getByText(new RegExp(`${props.spotRangeCounts?.[0].spots ?? 'feil'} plasser`))).toBeInTheDocument();
```
Comment (bakseter) [inline src/components/__tests__/happening-meta-info.test.tsx:169]: https://stackoverflow.com/questions/61480993/when-should-i-use-nullish-coalescing-vs-logical-or

**Review by boaanes — APPROVED** [review]
LGTM!

**Pull request merged**